overlord/ifacestate/handlers.go

```diff
@@ -349,6 +346,47 @@ func (m *InterfaceManager) undoDiscardConns(task *state.Task, _ *tomb.Tomb) erro
 	return nil
 }
 
+func getDynamicHookAttributes(task *state.Task) (map[string]interface{}, map[string]interface{}, error) {
+	var plugAttrs, slotAttrs map[string]interface{}
+
+	if err := task.Get("plug-dynamic", &plugAttrs); err != nil && err != state.ErrNoState {
+		return nil, nil, err
+	}
+	if err := task.Get("slot-dynamic", &slotAttrs); err != nil && err != state.ErrNoState {
+		return nil, nil, err
+	}
+	if plugAttrs == nil {
+		plugAttrs = make(map[string]interface{})
+	}
+	if slotAttrs == nil {
+		slotAttrs = make(map[string]interface{})
+	}
+
+	return plugAttrs, slotAttrs, nil
+}
+
+func setDynamicHookAttributes(task *state.Task, dynamicPlugAttrs map[string]interface{}, dynamicSlotAttrs map[string]interface{}) {
+	task.Set("plug-dynamic", dynamicPlugAttrs)
+	task.Set("slot-dynamic", dynamicSlotAttrs)
+}
+
+func markConnectHooksDone(connectTask *state.Task) error {
+	for _, t := range []string{"connect-plug-hook-task", "connect-slot-hook-task"} {
```

Comment:
This looks a bit suspect. How come we mark it done if it wasn't done? The connect task for a particular snap cannot possibly succeed if the snap isn't available right? I'm probably missing something.. can we please talk about it?

Reply:
This is only for auto-connect. In one of the older reviews @pedronis sugested that we ignore missing snaps at this point when auto-connecting in case the snap is gone by the time we execute this task. This egde case poses problem for interface hooks though as they will inevitably error out if they don't find given snap in the state.